src/ensembl/src/content/app/entity-viewer/gene-view/GeneView.tsx

```diff
@@ -180,17 +180,16 @@ const useGeneViewRouting = () => {
   const selectedTabs = useSelector(getSelectedGeneViewTabs);
 
   useEffect(() => {
-    if (previousGenomeId !== genomeId) {
-      if (viewInRedux && viewInRedux !== view) {
-        const url = urlFor.entityViewer({
-          genomeId,
-          entityId,
-          view: viewInRedux
-        });
-        dispatch(replace(url));
-      }
-    } else if (viewInRedux !== view) {
+    if (view && viewInRedux !== view) {
       dispatch(setGeneViewName(view));
+    } else {
+      const url = urlFor.entityViewer({
+        genomeId,
+        entityId,
+        view: viewInRedux
+      });
+
+      dispatch(replace(url));
```

Comment:
Could you explain the change please?

My thinking for the deleted version of the code was:
- View can be falsey (empty string or undefined, don't remember); and it actually is this way for the default transcripts list. After your change (`view && viewInRedux !== view`) I suspect that the view in redux will not be properly updated when switching to default transcripts view
- I also thought that the only case when the view in the url should be dictated by redux rather than by the url is when the user is switching the species.

If the reason for your change is to support a scenario such as this:
- switch to a non-default view in EntityViewer
- navigate to genome browser
- refresh the page so that redux state is reset
- navigate back to EntityViewer and see the view that you visited last time

then I am not even sure whether we want this behaviour (will need to consult with Andrea, of course)

Reply:
This change is just to handle the following scenario:

- switch to a non-default view in EntityViewer
- navigate to genome browser
- navigate back to EntityViewer and see the view that you visited last time

Also, with the previous implementation, when we revisit entity viewer without the view parameter, we do not know if the intended view is the default  `transcripts` or the view stored in redux. To handle this, I have added `?view=transcripts` to be the default view instead of an empty value.